### PR TITLE
cpu/sam0_common: add rtc_get_tamper_event() 

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

On SAM D5x/E5x only the 5 RTC tamper pins can wake the CPU from Deep Sleep.
Since this triggers a full reset, no callback can be executed. We have to read the `TAMPID` register to find out which event woke us from hibernation.

### Testing procedure

The new function was added to `tests/periph_gpio`

```
2021-03-11 22:28:03,257 # GPIO peripheral driver test
2021-03-11 22:28:03,257 # 
2021-03-11 22:28:03,262 # In this test, pins are specified by integer port and pin numbers.
2021-03-11 22:28:03,268 # So if your platform has a pin PA01, it will be port=0 and pin=1,
2021-03-11 22:28:03,271 # PC14 would be port=2 and pin=14 etc.
2021-03-11 22:28:03,271 # 
2021-03-11 22:28:03,276 # NOTE: make sure the values you use exist on your platform! The
2021-03-11 22:28:03,283 #       behavior for not existing ports/pins is not defined!
init_int 0 2 1
2021-03-11 22:28:05,350 #  init_int 0 2 1
2021-03-11 22:28:05,355 # GPIO_PIN(0, 2) successfully initialized as ext int

# test if the interrupt works
> 2021-03-11 22:28:07,214 #  INT: external interrupt from pin 2
pm set 0
2021-03-11 22:28:09,838 # pm set 0
2021-03-11 22:28:09,840 # CPU is entering power mode 0.
2021-03-11 22:28:09,843 # Now waiting for a wakeup event...
2021-03-11 22:28:17,833 # main(): This is RIOT! (Version: 2021.04-devel-980-g414c76-driver/lis2dh12-cleanup)
2021-03-11 22:28:17,835 # GPIO peripheral driver test
2021-03-11 22:28:17,835 # 
2021-03-11 22:28:17,841 # In this test, pins are specified by integer port and pin numbers.
2021-03-11 22:28:17,846 # So if your platform has a pin PA01, it will be port=0 and pin=1,
2021-03-11 22:28:17,849 # PC14 would be port=2 and pin=14 etc.
2021-03-11 22:28:17,850 # 
2021-03-11 22:28:17,855 # NOTE: make sure the values you use exist on your platform! The
2021-03-11 22:28:17,861 #       behavior for not existing ports/pins is not defined!
2021-03-11 22:28:17,861 # 
2021-03-11 22:28:17,864 # Woken by RTC tamper event: 4

# reboot to check if tamper event is not displayed again
2021-03-11 22:28:21,701 #  reboot
2021-03-11 22:28:21,710 # main(): This is RIOT! (Version: 2021.04-devel-980-g414c76-driver/lis2dh12-cleanup)
2021-03-11 22:28:21,712 # GPIO peripheral driver test
2021-03-11 22:28:21,713 # 
2021-03-11 22:28:21,718 # In this test, pins are specified by integer port and pin numbers.
2021-03-11 22:28:21,723 # So if your platform has a pin PA01, it will be port=0 and pin=1,
2021-03-11 22:28:21,726 # PC14 would be port=2 and pin=14 etc.
2021-03-11 22:28:21,727 # 
2021-03-11 22:28:21,732 # NOTE: make sure the values you use exist on your platform! The
2021-03-11 22:28:21,738 #       behavior for not existing ports/pins is not defined!
```

### CODE REVIEW

**Comment by kaspar030** [comment]
don't we have a wakesrc module?

**Comment by benpicco** [comment]
Which wakesrc module?

File: cpu/sam0_common/periph/rtc_rtt.c [hunk 1]
```diff
@@ -374,6 +384,26 @@ void rtc_tamper_enable(void)
...
 
+uint8_t rtc_get_tamper_event(void)
+{
+    uint32_t ret = RTC->MODE0.TAMPID.reg;
+
+    /* clear tamper event */
+    RTC->MODE0.INTFLAG.reg = RTC_MODE0_INTFLAG_TAMPER;
+    RTC->MODE0.TAMPID.reg  = ret;
+
+    return ret & 0x1F;
```
Comment (fjmolinas) [inline cpu/sam0_common/periph/rtc_rtt.c:395]: Why 0x1f?

**Comment by fjmolinas** [comment]
Looks good, one comment above though.

File: cpu/sam0_common/periph/rtc_rtt.c (hunk 1 above)
Comment (benpicco) [reply cpu/sam0_common/periph/rtc_rtt.c:395]: You're right, `RTC_TAMPID_TAMPID_Msk` is the better choice here.

**Review by fjmolinas — APPROVED** [review]
ACK, @benpicco provided test results.

**Pull request merged**